binding.cc

```diff
@@ -263,6 +274,246 @@ bare_sdl_get_event_key_scancode(
   return key->handle.scancode;
 }
 
+static uint32_t
+bare_sdl_open_audio_device(
+  js_env_t *env,
+  js_receiver_t,
+  uint32_t requested_device_id,
+  std::optional<uint32_t> format,
+  std::optional<int> channels,
+  std::optional<int> freq
+) {
+  int err;
+
+  SDL_AudioSpec *spec_ptr = nullptr;
+  if (format.has_value() && channels.has_value() && freq.has_value()) {
+    SDL_AudioSpec spec = {
+      .format = (SDL_AudioFormat) format.value(),
+      .channels = channels.value(),
+      .freq = freq.value()
+    };
+
+    spec_ptr = &spec;
```

Comment:
This isn't safe; `spec` is invalid after the scope exits.

Reply:
No need for a heap allocation, it just needs to be declared in the enclosing scope.

Reply:
✨fixed!